### DISCUSSION TITLE
Pull of repository only downloading readme file

### DISCUSSION THREAD

**Discussion by Marklack** [community/community#59897] [Pull Requests] [answered]
### Select Topic Area

Question

### Body

Hi I am having trouble pulling a repository.  It only will download the readme file, and there are several folders that it should be pulling. 

SSH, HTTPS does the same thing.  Zip will pull the folders but not the .csv embedded in the repository. 

Any help would be appreciated.

**Comment by leonden** [comment]
Based on your question I don't 100% understand your issue but I will share some thoughts I had while reading your post.

If you are pulling from the remote repository to your local copy, that could actually make sense. If you have created a _README.md_ on the webpage of the repository and committed the changes directly, with the next pull to your local repository it will only pull the _README.md_ if it was the only change.

If you have created directories that exist but don't include any files, they won't be integrated in your next commit until they have content. If you wish to always have your empty directories to be committed, you can simply create a _.gitkeep_ inside of your desired empty directory and you are good to go.

I hope that helped. Please let me know if you have any further information 😄

**Reply by Marklack** [reply]
So this is a repository I have access to .  I'm creating the repository space in gitbash on my machine. Then I pull the repository into my folder.  Is there a step I'm missing?

**Reply by leonden** [reply]
So you create a new local repository in Git Bash right? Do you reference it to a remote repository on GitHub? If not you could create a remote repo to reference your local directory to. Afterwards you can run

```sh
git init

git remote add origin <remote repository>

#optionally
git remote -v
```

in your local directory.

**Reply by Marklack** [reply]
![image](https://github.com/community/community/assets/11888192/96a6c49b-eb36-420f-a8b3-c04f1b7b31a9)

**Reply by Marklack** [reply]
![image](https://github.com/community/community/assets/11888192/40b11a69-0f89-4b92-9bcb-d46f1693b255)

**Reply by leonden** [reply]
So what I can see from your pictures is that in your file explorer the target directory is _bootcamp_ right? Then you can simply perform the shell commands I sent you in my previous answer and reference the directory _bootcamp_ to _DataVizz-Lesson-Plans_ which is your remote repository.

**Comment by tshenolo** [comment]
Hi there

Based on the screenshot you have provided try the following:

Check which branch you are currently on. I suspect main branch
git branch

Switch to the v1.2 branch and pull the changes
git checkout v1.2
git pull

Or you could merge changes on branch v.12 into the main branch on github.com then pull the changes.

Be careful because the message on your screenshot says:
This branch is 101 commits ahead, 7 commit behind main. (v1.2 Branch)

You made 7 changes to main branch that are not in v1.2 branch
and you made 101 changes in v1.2 that are not in main branch.

You will have to figure out how to merge the changes together.

Edit: 
Run the following command to view all the branches on your local repo:
git branch

look for the branch with name v1.2

Then do the following
git checkout BRANCH_NAME
git pull

**Reply by Marklack** [reply]
I'm a user of this and am trying to just clone it to my PC.  I can zip it, but the .csv files are all jumbled.  When I do a pull, that's all I get.  I installed LFS and still no help.

Incidentally I have a similar repository on git lab and it works fine.  Just got hub is my problem.![image](https://github.com/community/community/assets/11888192/af7eeca7-efd1-453c-8661-596c459ddae1)

**Reply by tshenolo** [reply]
Merge branch v1.2 to the main branch on GitHub then checkout the main branch then do a pull.

**Reply by Marklack** [reply]
Don't know I have the authority to make that merge.

**Reply by tshenolo** [reply]
Run the following command to view all the branches on your local repo:
git branch

look for the branch with name v1.2

Then do the following
git checkout <Branch with v1.2 in its name>
git pull

**Comment by Marklack** [comment]
That got it!  Thanks a ton!Sent from my iPhoneOn Jul 3, 2023, at 2:33 PM, Tshenolo Mos ***@***.***> wrote:﻿
Run the following command to view all the branches on your local repo:
git branch
look for the branch with name v1.2
Then do the following
git checkout 
git pull

**Reply by tshenolo** [reply, marked as answer]
I am glad I was able to help you with the issue. 
Please consider accepting my answer before closing the question. 
I have added an edit to my original answer.
Thanks
